internal/pkg/admission/reconciler.go

```diff
@@ -274,3 +322,45 @@ func (r *Reconciler) enablePolicyWebhook(
 
 	return nil
 }
+
+func (r *Reconciler) getClusterAdmissionPolicies(ctx context.Context, policyServer *policiesv1alpha2.PolicyServer) (policiesv1alpha2.ClusterAdmissionPolicyList, error) {
+	var clusterAdmissionPolicies policiesv1alpha2.ClusterAdmissionPolicyList
+	err := r.Client.List(ctx, &clusterAdmissionPolicies, client.MatchingFields{constants.PolicyServerIndexKey: policyServer.Name})
+
+	return clusterAdmissionPolicies, err
+}
+
+func (r *Reconciler) deletePendingClusterAdmissionPolicies(ctx context.Context, clusterAdmissionPolicies policiesv1alpha2.ClusterAdmissionPolicyList) error {
+	for _, policy := range clusterAdmissionPolicies.Items {
+		if policy.DeletionTimestamp != nil {
+			if policy.Spec.Mutating {
+				mutatingWebhook := &admissionregistrationv1.MutatingWebhookConfiguration{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: policy.Name,
+					},
+				}
+				err := r.Client.Delete(context.Background(), mutatingWebhook)
+				if err != nil && !apierrors.IsNotFound(err) {
+					return err
+				}
+			} else {
+				validatingWebhook := &admissionregistrationv1.ValidatingWebhookConfiguration{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: policy.Name,
+					},
+				}
+				err := r.Client.Delete(context.Background(), validatingWebhook)
+				if err != nil && !apierrors.IsNotFound(err) {
+					return err
+				}
+			}
+			patch := policy.DeepCopy()
+			patch.ObjectMeta.Finalizers = nil
```

Comment:
Same here, I think we should only remove our finalizer, letting other finalizers alone in case there are.